### PR TITLE
Rewrite queue drag&drop

### LINKED ISSUE #209: Rewrite QueueList
QueueList drag and drop is poorly done. Rewrite it.

- [ ] Instead of placeholders, use a simple line, more simple to deal with
- [ ] Use real indexes instead of calculating them by a tricky formula
- [ ] Fix component styling and paddings

### PR DESCRIPTION
**Pull request by KeitIG**
fix #209 

- More fluid and trackable drag&drop system
- Fix drop on higher position in the list
- Style update: easier to manage drag/dragover actions

@YurySolovyov, about the review: just review the code itself, not the system, I prefered not to implement react-dnd for now, no time for that.

### CODE REVIEW

File: src/js/components/Header/QueueList.react.js [hunk 1]
```diff
@@ -108,42 +105,38 @@ export default class QueueList extends Component {
 
         const draggedTrack     = this.state.draggedTrack;
         const draggedOverTrack = this.state.draggedOverTrack;
+        const dragPosition     = this.state.dragPosition;
+
+        // If someone has a better idea...
+        const offset = (dragPosition === 'below' ? 1 : 0) + (draggedOverTrack < draggedTrack || (draggedOverTrack === draggedTrack && dragPosition === 'above') ? 1 : 0);
```
Comment (YurySolovyov) [inline src/js/components/Header/QueueList.react.js]: can you break this line?

File: src/js/components/Header/QueueList.react.js [hunk 2]
```diff
@@ -108,42 +105,38 @@ export default class QueueList extends Component {
...
+
+        const newQueue = [...queue];
+
+        // remove draggedTrack
+        const movedTrack = newQueue.splice(queueCursor + 1 + draggedTrack, 1)[0];
```
Comment (YurySolovyov) [inline src/js/components/Header/QueueList.react.js]: Can you explain a bit what is calculated here?

File: src/js/components/Header/QueueList.react.js (hunk 1 above)
Comment (KeitIG) [inline src/js/components/Header/QueueList.react.js]: Yeah sure.

File: src/js/components/Header/QueueList.react.js (hunk 2 above)
Comment (KeitIG) [inline src/js/components/Header/QueueList.react.js]: The whole queue is sent to `Queue`. So the correct moved track position in the queue is `queueCursor + 1 (for current track playing) + draggedTrack (dragged track index`)

Comment (YurySolovyov) [inline src/js/components/Header/QueueList.react.js]: can you move that index to var? I mean `queueCursor + 1 + draggedTrack`

**Comment by YurySolovyov** [comment]
This LGTM with comments addressed.

**Comment by KeitIG** [comment]
If it looks good now, let's ship it.

**Pull request merged**